pkg/controller/tailoredprofile/tailoredprofile_controller.go

```diff
@@ -182,33 +177,42 @@ func (r *ReconcileTailoredProfile) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	// Check if this ConfigMap already exists
-	found := &corev1.ConfigMap{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: tpcm.Name, Namespace: tpcm.Namespace}, found)
-	if err != nil && errors.IsNotFound(err) {
-		// update status
-		err = r.updateTailoredProfileStatusReady(instance, tpcm)
+	return r.ensureOutputObject(instance, tpcm, reqLogger)
+}
+
+// validates the given TailoredProfile. true means that we can continue since the
+// tailored profile is valid, false means that we can't.
+func (r *ReconcileTailoredProfile) validateTailoredProfile(tp *compliancev1alpha1.TailoredProfile) (bool, error) {
+	// Validate TailoredProfile
+	if tp.Spec.Extends == "" {
+		err := r.updateTailoredProfileStatusError(tp, fmt.Errorf(".spec.extends can't be empty"))
 		if err != nil {
-			fmt.Printf("Couldn't update TailoredProfile status: %v\n", err)
-			return reconcile.Result{}, err
+			return false, err
 		}
+		// don't return an error in the reconciler. The error will surface via the CR's status
+		return false, nil
+	}
 
-		// create CM
-		reqLogger.Info("Creating a new ConfigMap", "ConfigMap.Namespace", tpcm.Namespace, "ConfigMap.Name", tpcm.Name)
-		err = r.client.Create(context.TODO(), tpcm)
+	switch tp.Spec.OutputType {
+	case compliancev1alpha1.ConfigMapOutput: // Do nothing, we're good
+	case "":
+		tpCopy := tp.DeepCopy()
+		tpCopy.Spec.OutputType = compliancev1alpha1.ConfigMapOutput
```

Comment:
Isn't this update a noop?

Reply:
I did this before I added the Enum... not sure if I should keep it or not. All this does is ensure that we always have a valid outputType (sets a default, updates and the updated one will be queued)